src/blas/backends/syclblas/CMakeLists.txt

```diff
@@ -0,0 +1,197 @@
+#==========================================================================
+#  Copyright (C) Codeplay Software Limited
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  For your convenience, a copy of the License has been included in this
+#  repository.
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+#=========================================================================
+
+set(LIB_NAME onemkl_blas_syclblas)
+set(LIB_OBJ ${LIB_NAME}_obj)
+
+option(SYCLBLAS_TUNING_TARGET "Set a TUNING_TARGET for SYCL-BLAS" "")
+
+# Parse compiler flags and return a list of SYCL targets
+# The list is empty if no targets are set
+function(get_sycl_targets FLAGS)
+  string(REGEX MATCH "-fsycl-targets=[^ ]*" SYCL_TARGETS_FLAG "${FLAGS}")
+  string(REPLACE "-fsycl-targets=" "" SYCL_TARGETS "${SYCL_TARGETS_FLAG}")
+  string(REPLACE "," ";" SYCL_TARGETS "${SYCL_TARGETS}")
+  set(SYCL_TARGETS ${SYCL_TARGETS} PARENT_SCOPE)
+endfunction(get_sycl_targets)
+
+# SYCL-BLAS supports tuning for some device types, but can only be compiled
+# for one at a time currently. Work out which device to tune for based on the
+# DPC++ target triple specified via -fsycl-targets
+if(TARGET ONEMKL::SYCL::SYCL)
+  get_target_property(ONEMKL_COMPILE_OPTIONS ONEMKL::SYCL::SYCL INTERFACE_COMPILE_OPTIONS)
+endif()
+get_sycl_targets("${ONEMKL_COMPILE_OPTIONS}")
+list(LENGTH SYCL_TARGETS NUM_TARGETS)
+if(NUM_TARGETS EQUAL 0)
+  get_sycl_targets("${CMAKE_CXX_FLAGS}")
+  list(LENGTH SYCL_TARGETS NUM_TARGETS)
+endif()
+
+if(SYCLBLAS_TUNING_TARGET)
+  # Allow the user to manually enable a specific device type without relying on the sycl target.
+  if(SYCLBLAS_TUNING_TARGET STREQUAL "INTEL_CPU")
+    # Allow the user to enable only the INTEL_CPU backend even though SYCL-BLAS cannot be tuned for INTEL_CPU.
```

Comment:
Why it cannot be tuned for INTEL_CPU?

Reply:
The SYCL-BLAS readme says:
> By default SYCL-BLAS library is tuned for CPU. Use that flag to tune it for a target (highly recommended for performance). The supported targets are: INTEL_GPU, AMD_GPU, ARM_GPU, RCAR

It's not quite the same wording I have used but it conveys the idea that the user should not set any `TUNING_TARGET` to run on CPU devices. Let me know if you prefer that I reword the comment in `CMakeLists.txt`.
I am trying to keep the documentation in oneMKL minimal and generic as there is a risk it won't be kept up to date.

Reply:
I see. I'd expect "tuned for CPU" means functional and performance support, so this sentence confused me. 

I don't think users read comments for internal `CMakeLists.txt` so the comment is fine, but having explicit statement about CPU is not performance target in oneMKL or in SYCL BLAS documentation could help to make it clear to users.

Reply:
Hi @mkrainiuk,
I have put up this PR on SYCL-BLAS (https://github.com/codeplaysoftware/sycl-blas/pull/404) to update the README to address the confusion above. Let me know if this change clarifies the confusion or not? Thanks.

Reply:
Thanks @muhammad-tanvir-1211! Yes, it makes it more clear to me.